src/type-syncer.ts

```diff
@@ -88,16 +91,19 @@ export function createTypeSyncer(
     filePath: string,
     file: IPackageFile | null,
     allTypings: Array<ITypeDefinition>,
-    opts: ISyncOptions
+    opts: ISyncOptions,
+    dryRun: boolean
   ): Promise<ISyncedFile> {
-    const ignore = opts.ignore || {}
+    const { ignoreDeps = [], ignorePackages = [] } = opts
+
     file = file || (await packageJSONService.readPackageFile(filePath))
-    const allPackages = [
-      ...getPackagesFromSection(file.dependencies, ignore.deps),
-      ...getPackagesFromSection(file.devDependencies, ignore.dev),
-      ...getPackagesFromSection(file.optionalDependencies, ignore.optional),
-      ...getPackagesFromSection(file.peerDependencies, ignore.peer),
-    ]
+    const allPackages = file
+      ? Object.values(IDependencySection)
+          .filter((dep) => !ignoreDeps.includes(dep))
+          .map((dep) => getDependenciesBySection(file!, dep))
+          .map((section) => getPackagesFromSection(section, ignorePackages))
```

Comment:
You can consolidate the 2 `maps` so we don't iterate more than we have to, and you can use `filterMap` from `util` to do the filtering and mapping in a single pass.